lightning/src/ln/channelmanager.rs

```diff
@@ -3166,6 +3203,9 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 {
 	/// Updates channel state based on transactions seen in a connected block.
 	pub fn block_connected(&self, header: &BlockHeader, txdata: &TransactionData, height: u32) {
+		// Note that we MUST NOT end up calling methods on self.chain_monitor here - we're called
+		// during initialization prior to the chain_monitor being fully configured in some cases.
+		// See the docs for `ChannelManagerReadArgs` for more.
```

Comment:
Hm, is there a way to put some asserts on `chain_monitor` state that it's the same at the beginning and end of these functions? Seems safer than just leaving comments.

Reply:
Yea, I thought about that, we would have to wrap the entire `chain_monitor` in a wrapper struct that will check a boolean before calling the inner method. I figured it wasn't worth the complexity, but I'm open to other opinions.

Reply:
This does seem to indicate a need to refactor the code such that `chain_monitor` cannot be called in certain scenarios. I don't have a concrete suggestion, however.

Reply:
If we wanted to go there, we could have deserialization build some intermediate object which you can only connect/disconnect blocks on, then you can tell it you're done and get the full `ChannelManager`. Not sure how that would interact with #800, though.

Reply:
No need to address now, but as part of #800 we may want to catalog all that `ChannelManager` is "managing" and identify suitable abstractions where possible. :)

Reply:
Added a comment describing this on 800.

Reply:
> Yea, I thought about that, we would have to wrap the entire `chain_monitor` in a wrapper struct that will check a boolean before calling the inner method. I figured it wasn't worth the complexity, but I'm open to other opinions.

Hm, could you retrieve the `ChainMonitor`'s list of `ChannelMonitor` outpoints and latest update ID's at the beginning of the function, then ensure those are the same at the end of the function?